### ISSUE TITLE
ovh_ip_reverse fails to unmarshal the response because it receives an array of ip addresses

### ISSUE THREAD

**Issue by berbiche** [ovh/terraform-provider-ovh#208]
### Terraform Version

```console
Terraform v0.14.11
+ provider registry.terraform.io/backblaze/b2 v0.2.1
+ provider registry.terraform.io/carlpett/sops v0.6.3
+ provider registry.terraform.io/cloudflare/cloudflare v2.23.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/null v3.1.0
+ provider registry.terraform.io/ovh/ovh v0.14.0
```

### Affected Resource(s)

- ovh_ip_reverse

### Terraform Configuration Files

```hcl
data "ovh_vps" "something" {
  service_name = "<elided>"
}

locals {
  ipv4_addresses = toset([for s in data.ovh_vps.something.ips : s if !can(cidrnetmask("${s}/128"))])
  ipv6_addresses = toset([for s in data.ovh_vps.something.ips : s if can(cidrnetmask("${s}/128"))])
}

resource "ovh_ip_reverse" "something_ipv4" {
  for_each = local.ipv4_addresses
  ip = "${each.value}/32"
  reverse = "example.com."
}

resource "ovh_ip_reverse" "something_ipv6" {
  for_each = local.ipv6_addresses
  ip = "${each.value}/128"
  reverse = "example.com."
}
```

### Debug Output

Sorry I did not respect the guidelines here but the debug output is short and the bug has been precisely defined in the section "Actual behavior".

```
2021-07-04T01:01:31.892-0400 [INFO]  plugin.terraform-provider-ovh_v0.14.0: 2021/07/04 01:01:31 [DEBUG] OVH API Request Details:
---[ REQUEST ]---------------------------------------
GET /1.0/ip/<elided>%2F32/reverse/ HTTP/1.1
Host: ca.api.ovh.com
User-Agent: Go-http-client/1.1
<elided>

2021-07-04T01:01:32.063-0400 [INFO]  plugin.terraform-provider-ovh_v0.14.0: 2021/07/04 01:01:32 [DEBUG] OVH API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
<elided>

[
 "XX.XXX.XX.XXX"
]
```

### Panic Output

This used to crash Terraform in release 0.11 but after upgrading to 0.14 there's a useful error message:

```
Error: calling /ip/<elided>%2F32/reverse/:
	 json: cannot unmarshal array into Go value of type ovh.OvhIpReverse
```

### Expected Behavior

Per the title, when looking up the state of `ovh_ip_reverse`, it should not fail to validate.

### Actual Behavior

The code expects the API to return a flat object but it returns an array.
https://github.com/ovh/terraform-provider-ovh/blob/54b29545df8fc82899e9d462c7c46a54a15d463e/ovh/resource_ip_reverse.go#L102-L120

As you can see from my configuration file, I did not specify the `ipreverse` field, the code then uses the API endpoint `ip/<ip>/reverse` and according to the [official documentation](https://api.ovh.com/console/#/ip/%7Bip%7D/reverse#GET) this endpoint returns a list of ips.

~~The problem is "resolved" if I set `ipreverse` to the value of `ip` without the mask.~~
Though I need to delete the resource from my state because the state does not have an `ipreverse`.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Nothing

### References

Nothing

**Comment by yanndegat** [comment]
hi @berbiche 

thank you for this very clean report. it greatly helps the debugging.

While trying to reproduce & fix the issue, i noticed several flaws in the resource implementation, so i rewrote it
following the same patterns we use for other resources.

the most notable (and breaking) changes are that now 

- ipreverse is renamed ip_reverse to follow the naming convention
- ip_reverse is mandatory.

in a global manner, we'd like the provider to mimic the API behaviour & design, without embedding custom logic.
this would eventually allow us to generate the provider from the api definition.

i'ill try to release a new version which this new version included soon

**Comment by berbiche** [comment]
Thanks the fix in #209 indeed work.
